Expand the critial section of the readers/writers.

e-harvey opened the pull request:
- Added volatile writes to both the readers and writers.

@hppritcha @ztiffany @jswaro @jshimek @tonyzinger @chuckfossen @sungeunchoi here are some results from these updated tests.
[lock-tests-glibc-2.11.3-2r-2w-2to20rw.pdf](https://github.com/ofi-cray/cray-tests/files/362051/lock-tests-glibc-2.11.3-2r-2w-2to20rw.pdf)
[lock-tests-glibc-2.11.3-128r-128w-2to14rw.pdf](https://github.com/ofi-cray/cray-tests/files/362054/lock-tests-glibc-2.11.3-128r-128w-2to14rw.pdf)

Edit: These results were collected using 9ab2c72.

Comparing the results above to #22 it appears that mutexes are an optimal choice when there is _very_ _little_ IO/CPU work going on in the critical section, how little work will not be clear without further testing.

As shown by the results listed above in this PR, rwlocks appear to perform better when there are _more_ IO/CPU intensive operations within the critical section.

Commit:
Expand the critial section of readers/writers.

       - Added volatile writes to both the readers and
       writers.

Signed-off-by: Evan Harvey <eharvey@cray.com>

sungeunchoi commented:
Why does the reader read from a local and the writer write to a global?

e-harvey commented:
The "tmp" local variable in the reader is just to waste some time within that critical section.  The reader still reads the global "np" and writes the contents of "np" to "p".

Similarly, the writer wastes time with the global "np" variable by incrementing it 128 times.

On Wed, Jul 13, 2016 at 5:58 PM -0500, "Sung-Eun Choi" <notifications@github.com<mailto:notifications@github.com>> wrote:

Why does the reader read from a local and the writer write to a global?

## 

You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHubhttps://github.com/ofi-cray/cray-tests/pull/24#issuecomment-232511023, or mute the threadhttps://github.com/notifications/unsubscribe/AHET8oYqpz8cDQzFD2_zFF1ZUKQWduG-ks5qVW13gaJpZM4JLptk.

sungeunchoi commented:
I guess I missed that the test actually does reads and writes in the loop.  That's not really necessary, so to keep it simple I'd change it to just to the volatile writes to a local variable in both the reader and writer just to kill time.

Commit:
Implemented PR 24 feedback.

	  - Removed functionality checks.
	  - Added voltile writes to both reader and writer.

Signed-off-by: Evan Harvey <eharvey@cray.com>

sungeunchoi commented:
👍
